web3/utils/abi.py

```diff
@@ -1,21 +1,242 @@
+import functools
 from typing import (
+    Any,
     List,
+    Optional,
+    Sequence,
     Union,
+    cast,
 )
 
+from eth_typing import (
+    ABIFunctionInfo,
+)
+from eth_utils.abi import (
+    function_abi_to_4byte_selector,
+    get_aligned_abi_inputs,
+    get_normalized_abi_inputs,
+)
+from eth_utils.encoding import (
+    get_default_codec,
+)
+from eth_utils.hexadecimal import (
+    encode_hex,
+)
+from eth_utils.toolz import (
+    pipe,
+)
+from eth_utils.types import (
+    is_text,
+)
+from typing_extensions import (
+    deprecated,
+)
+
+from web3._utils.abi import (
+    abi_to_signature,
+    filter_by_argument_count,
+    filter_by_encodability,
+    filter_by_name,
+)
+from web3._utils.contracts import (
+    extract_argument_types,
+)
+from web3.exceptions import (
+    MismatchedABI,
+)
 from web3.types import (
+    ABI,
     ABIEvent,
     ABIFunction,
 )
 
+DEPRECATION_MESSAGE = (
+    "`{0}` is deprecated and will be removed in a future release. Use " "`{1}` instead."
+)
+
 
+@deprecated(
+    DEPRECATION_MESSAGE.format(
+        "get_abi_input_names", "eth_utils.abi.get_abi_output_names"
+    )
+)
+@deprecated("Use eth_utils.abi.get_abi_input_names")
 def get_abi_input_names(abi: Union[ABIFunction, ABIEvent]) -> List[str]:
+    """
+    Return the ``input`` names for an ABI function or event.
+    """
     if "inputs" not in abi and abi["type"] == "fallback":
         return []
     return [arg["name"] for arg in abi["inputs"]]
 
 
+@deprecated(
+    DEPRECATION_MESSAGE.format(
+        "get_abi_output_names", "eth_utils.abi.get_abi_output_names"
+    )
+)
 def get_abi_output_names(abi: Union[ABIFunction]) -> List[str]:
     if "outputs" not in abi and abi["type"] == "fallback":
         return []
     return [arg["name"] for arg in abi["outputs"]]
+
+
+def get_function_info(
```

Comment:
It doesn't seem like we deprecated `web3.utils.abi.py` as the newsfragment states so I'm confused by that in the newsfragment. My lingering question remains, are we re-exposing `eth_utils` or asking users to import different utils from different places? I lean towards the former for a better UX of centralized utils but I'm not sure I'm caught up on the conversation around it.

Reply:
The newsfragment is outdated, sorry about that.